docs/DEVELOPMENT_GUIDE.md

```diff
@@ -0,0 +1,111 @@
+# Development Guide
+
+## How to Contribute
+
+We'd love to accept your patches and contributions to this project. There are
+just a few small guidelines you need to follow.
+
+### Submit Contributor License Agreement
+If you are not a member of the Google Genomics Organization, you will need to
+sign a Contributor License Agreement before we can review your pull request.
+Please visit
+[https://cla.developers.google.com/](https://cla.developers.google.com/clas)
+to sign.
+
+### Code reviews
+
+All submissions, including submissions by project members, require review. We
+use GitHub pull requests for this purpose. Consult
+[GitHub Help](https://help.github.com/articles/about-pull-requests/) for more
+information on using pull requests.
+
+
+## Setup
+
+### Fork the repository on Github
+Visit the [gcp-variant-transforms repository](https://github.com/googlegenomics/gcp-variant-transforms)
+to create your own fork of the repository.
+
+### Setup dev environment
+
+#### Clone the forked repository
+```
+$ git clone git@github.com:<username>/gcp-variant-transforms.git
+$ cd gcp_variant_transforms
+```
+You will also googlegenomics repository as a remote so that you can easily pull
+changes later.
+```
+$ git remote add git@github.com:googlegenomics/gcp-variant-transforms.git
+```
+
+#### Setup virtualenv
+```
+$ sudo apt-get install virtualenv
+$ virtualenv venv
+$ . venv/bin/activate
+```
+
+#### Install dependences
+```
+$ python setup.py install
+```
+
+## Making Changes
+### Create a branch in your forked repository
+```
+git checkout -b <branch-name> origin/master
+```
+
+Running this command will create a branch named `<branch-name>` and switch
+you to it.
+
+### Testing
+To run all tests:
+
+```bash
+python setup.py test
+```
+
+To run a specific test:
+```bash
+python setup.py test -s <module>.<test class>.<test method>
+```
+
+### Pushing changes to your fork's branch
+To push changes to your forked branch, you can run:
+```
+git add -p
+```
+This will allow you to browse through changes since your last commit and filter
+the exact changes that you want to commit. You can then run:
+```
+git commit -m "<commit message>"
+git push
```

Comment:
does this work? I thought you had to specify with "git push origin -u"

Reply:
git push will work if upstream is configured to be origin/master. Otherwise you can run `git push origin <branch-name>`. The `-u` parameter will set your upstream but after doing it once you can just run `git push`